common/src/main/java/net/opentsdb/data/TimeSeriesId.java

```diff
@@ -0,0 +1,122 @@
+ * An identifier for a time series. The identity can be as simple as the alias
+ * or a combination of namespace, metrics, tags, etc.
+ * <p>
+ * Most of the fields return a list or map. These represent the aggregate of all
+ * time series included in the sourcing of this series. This is applicable to
+ * synthetic series when multiple time series are aggregated into one series.
+ * <p>
+ * In general, if a raw time series is the source then only one metric and
+ * an optional set of tags should be present.
+ * 
+ * @since 3.0
+ */
+public interface TimeSeriesId extends Comparable<TimeSeriesId> {
+  /**
+   * @return True if the fields are encoded using a format specified by the 
+   * storage engine.
+   */
+  public boolean encoded();
+  
+  /**
+   * A simple string for identifying the time series. The alias may be null or
+   * empty. If a value is present, the alias must be unique within a set of
+   * time series.
+   * <p>
+   * Invariant: If all other fields for the identity are empty, then the alias
+   * must be non-null and non-empty.
+   * 
+   * @return A string or null if not used.
+   */
+  public String alias();
+  
+  /**
+   * A list of namespace identifiers included in the sources for this time series.
+   * May be empty if namespaces are not in use for the platform.
+   *  
+   * @return A non-null list if zero or more namespace identifiers.
+   */
+  public List<String> namespaces();
+  
+  /**
+   * A list of metric names included in the sources for this time series.
+   * If this list is empty, check {@link #metrics()} for data.
+   * <p>
+   * Invariant: If this list is empty (meaning the series or set of series is 
+   * not associated with a specific metric) then the {@link #alias()} must be 
+   * non-null and unique.
+   *  
+   * @return A non-null list of zero or more metric names.
+   */
+  public List<String> metrics();
+  
+  /**
+   * A list of tag name and value pairs included in the sources for this time 
+   * series. If the underlying storage system does not support tags or none of
+   * the source time series had tags pairs in common, this list may be empty.
+   * The map key represents a tag name (tagk) and the map value represents a 
+   * tag value (tagv).
+   * <p>
+   * Invariant: Each tag pair must appear in every source time series.
+   * 
+   * @return A non-null map of zero or more tag pairs.
+   */
+  public Map<String, String> tags();
+
+  /**
+   * A list of tag names (tagk) that were represented in every source series
+   * but had one or more differing tag values (tagv). This list may be empty if
+   * - The underlying store does not support tags.
+   * - All of the source time series shared the same tag pairs.
+   * - All of the source time series had disjoint tag sets.
+   * <p>
+   * Invariant: Each tag name must be present in every time series if it appears
+   * in this list.
+   * 
+   * @return A non-null list of zero or more tag names.
+   */
+  public List<String> aggregatedTags();
```

Comment:
I think it would be a good idea to have one interface for data and one interface for the response containing aggregated tags etc.. That way we can use the basic data interface in a store like metron.

Reply:
Hmm, We could do that. This one has a lot of bits for the query ID vs the raw data ID.